source/solvers/vof.cc

```diff
@@ -367,14 +380,24 @@ VolumeOfFluid<dim>::calculate_volume(int id_fluid_monitored)
                   case 0:
                     {
                       if (q_scalar_values[q] < 0.5)
-                        volume +=
-                          fe_values_vof.JxW(q) * (1 - q_scalar_values[q]);
+                        {
+                          this->volume_monitored +=
+                            fe_values_vof.JxW(q) * (1 - q_scalar_values[q]);
+                          this->mass_monitored +=
+                            this->volume_monitored * density_0[q];
```

Comment:
yes, `q_scalar_values` is the VOF phase indicator. I used the same name `q_scalar_values` as in `calculate_L2_error` where we also have `fe_values_vof.get_function_values(this->present_solution, q_scalar_values);`, but I see that in `assemble_L2_projection_interface_sharpening` it is called `phase_values`. I will change that for the sake of clarity!

Reply:
as for the mass calculation, I think I got your point. I wanted to save some calculation time, as I am only interested in the `fluid_monitored` in the end, but it is true that because of diffusion we still have a bit of the fluid monitored outside the domain.

To be sure I understand correctly: I should keep the `volume_monitored` computation behind the interface only (it was a previous PR), or I should also compute it on the entire domain (to assess the volume part that is being diffused)? I would go with the later (compute both mass and volume on the entire domain), considering your comment

Reply:
I changed it on the last push! The more I think about it, the more I wonder why I did it differently to start with ^^" Thank you for pointing this out!

Reply:
No worries, it's a bit confusing because we always see the fluid as in the fluid above the 0.5 iso value, but in reality, the mass conserved is the integral of the phase value everywhere!